internal/agent/device/spec/manager.go

```diff
@@ -300,6 +343,10 @@ func (s *manager) Rollback(ctx context.Context, opts ...RollbackOption) error {
 
 	// rollback in-memory cache current == desired
 	s.cache.update(Desired, current)
+
+	// Log successful rollback to audit log
+	s.logAudit(ctx, current, failedDesiredVersion, current.Version(), audit.AuditResultSuccess, audit.AuditReasonRollback, audit.AuditTypeRollback)
```

Comment:
at line 332 we are copying the current to the desired so this is technically the desired state that has changed.
```suggestion
	s.logAudit(ctx, current, failedDesiredVersion, current.Version(), audit.AuditResultSuccess, audit.AuditReasonRollback, audit.AuditTypeDesired)
```

Reply:
:heavy_check_mark: resolved in f74534b270b3825702f72989b34afb64cb057bdf